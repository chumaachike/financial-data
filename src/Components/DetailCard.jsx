import React from 'react';
import PropTypes from 'prop-types';
import './companylist.css';

function DetailCard({
  symbol,
  reportedCurrency,
  revenue,
  grossProfit,
  dateAccepted,
  calendarYear,
  costOfRevenue,
  interestIncome,
}) {
  return (
    <div className="details-card">
      <h3>
        {calendarYear}
        {' '}
        Details
      </h3>
      <div className="details-element">
        <p>Symbol:</p>
        <p>{symbol}</p>
      </div>
      <div className="details-element">
        <p>Reported Currency:</p>
        <p>{reportedCurrency}</p>
      </div>
      <div className="details-element">
        <p>Revenue:</p>
        <p>{revenue}</p>
      </div>
      <div className="details-element">
        <p>Gross Profit:</p>
        <p>{grossProfit}</p>
      </div>
      <div className="details-element">
        <p>Date Accepted:</p>
        <p>{dateAccepted}</p>
      </div>
      <div className="details-element">
        <p>Cost of Revenue:</p>
        <p>{costOfRevenue}</p>
      </div>
      <div className="details-element">
        <p>Interest Icome:</p>
        <p>{interestIncome}</p>
      </div>
    </div>
  );
}

export default DetailCard;

DetailCard.propTypes = {
  symbol: PropTypes.string.isRequired,
  reportedCurrency: PropTypes.string.isRequired,
  revenue: PropTypes.number.isRequired,
  grossProfit: PropTypes.number.isRequired,
  dateAccepted: PropTypes.string.isRequired,
  calendarYear: PropTypes.string.isRequired,
  costOfRevenue: PropTypes.number.isRequired,
  interestIncome: PropTypes.number.isRequired,
};
