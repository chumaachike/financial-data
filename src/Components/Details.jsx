import React from 'react';
import { useSelector } from 'react-redux';
import { v4 as uuid } from 'uuid';
import { NavLink } from 'react-router-dom';
import DetailCard from './DetailCard';

function Details() {
  let renderedItem = 'Loading';
  const companyDetails = useSelector((state) => state.CompanyDetails);
  if (companyDetails['Error Message']) {
    renderedItem = <p style={{ color: 'black' }}>{companyDetails['Error Message']}</p>;
  } else {
    renderedItem = companyDetails.map(
      (detail) => (
        <DetailCard
          key={uuid()}
          symbol={detail.symbol}
          reportedCurrency={detail.reportedCurrency}
          revenue={detail.revenue}
          grossProfit={detail.grossProfit}
          dateAccepted={detail.acceptedDate}
          calendarYear={detail.calendarYear}
          costOfRevenue={detail.costOfRevenue}
          interestIncome={detail.interestIncome}
        />
      ),
    );
  }
  return (
    <>
      <h1 className="header">Details</h1>
      <NavLink to="/" className="navlink">Click to go back to all stocks</NavLink>
      <div className="details">
        {renderedItem}
      </div>
    </>
  );
}

export default Details;
