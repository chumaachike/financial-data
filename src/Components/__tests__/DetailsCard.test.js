import React from 'react';
import renderer from 'react-test-renderer';
import Details from '../Details';
import DetailCard from '../DetailCard';

const Container = () => {
  <Details>
    <DetailCard />
  </Details>;
};

it('DetailCard renders correctly', () => {
  const tree = renderer.create(<Container />).toJSON();
  expect(tree).toMatchSnapshot();
});
