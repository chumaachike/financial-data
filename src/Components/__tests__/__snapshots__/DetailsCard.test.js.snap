// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailCard renders correctly 1`] = `null`;
